app/containers/ArticleApp/components/ArticleList.js

```diff
@@ -0,0 +1,207 @@
+/* @flow */
+
+import React from 'react';
+import Relay from 'react-relay';
+import ArticleItem from './ArticleItem';
+import CreateArticleModal from "./CreateArticleModal.react";
+import Sortable from "components/Sortable.react";
+import AddArticleMutation from "../mutations/AddArticleMutation";
+
+type ArticleEdge = {
+  node: {
+    id: string,
+  },
+};
+
+type Props = {
+  user: {
+    id: any,
+  },
+  relay: {
+    commitUpdate: Function,
+  },
+  articles: {
+    edges: Array<ArticleEdge>, // eslint-disable-line
+  },
+};
+
+const SORT = {
+  AUTHOR: 1,
+  TITLE: 2,
+};
+
+const SORT_KEY = {
+  [SORT.AUTHOR]: 'author',
+  [SORT.TITLE]: 'title',
+};
+
+/**
+ * Article list component
+ * Parent component: ArticleApp
+ * Child component: ArticleItem
+ */
+class ArticleList extends React.Component {
+
+  props: Props;
+
+  static getSortValue(obj, col) {
+    switch (col) {
+      // Use toUpperCase() to ignore character casing
+      case SORT.AUTHOR:
+      case SORT.TITLE:
+        return obj[SORT_KEY[col]].toUpperCase();
+      default:
+        throw new Error('Invalid sort');
+    }
+  }
+
+  _toggleModal() {
+    this.setState({
+      showModal: !this.state.showModal,
+    });
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      sort: {
+        direction: null,
+        col: null,
+      },
+      showModal: false,
+    };
+    this._sort = this._sort.bind(this);
+    this._toggleModal = this._toggleModal.bind(this);
+    this._onAdd = this._onAdd.bind(this);
+  }
+
+  /**
+   * Sort the list of articles based on the way the user selected.
+   */
+  _sort(direction, col) {
+    this.setState({
+      sort: {direction, col,},
+    });
+  }
+
+  /**
+   * Event handler for when the user has successfully added an article.
+   */
+  _onAdd(article) {
+    this.props.relay.commitUpdate(
+      new AddArticleMutation({
+        user: this.props.user,
+        ...article,
+      })
+    );
+
+    this.setState({
+      showModal: false,
+    });
+  }
+
+  render() {
+    const articles = this._getSortedArticles();
+
+    return (
+      <div>
+        <h4>
+          Articles &nbsp;
+          <a
+            role="link"
+            tabIndex={0}
+            style={{cursor: 'pointer'}}
+            onClick={this._toggleModal}
+          >+</a>
+        </h4>
+        <table>
+          <thead>
+          <tr>
+            <th>
+              Author &nbsp;
+              <Sortable col={SORT.AUTHOR} onSort={this._sort}/>
+            </th>
+            <th>
+              Title &nbsp;
+              <Sortable col={SORT.TITLE} onSort={this._sort}/>
+            </th>
+          </tr>
+          </thead>
+          <tbody>
+          {articles.map((edge, key) =>
+            <ArticleItem
+              key={key}
+              article={edge}
+            />
+          )}
```

Comment:
Since this render function is already pretty long, you could create a function that just returns lines 131 to 136 in a function. You could do the same with the table that starts on line 117, its up to you. Having long render functions isn't necessarily a bad thing, just something to think about.

Reply:
I definitely agree that splitting it up would be preferable here. I will leave it as-is since it's just a training project but will remember that next time.

Reply:
I believe that this sort function sorts the column alone. Aren't the table contents supposed to be sorted based on the column? As in, the rows should change based on how this column gets sorted.

Reply:
Please add comments for each function too.